xdsl/utils/mlir_lexer.py

```diff
@@ -258,12 +258,17 @@ def _consume_regex(self, regex: re.Pattern[str]) -> re.Match[str] | None:
         return match
 
     _whitespace_regex = re.compile(r"((//[^\n]*(\n)?)|(\s+))*", re.ASCII)
+    _first_whitespace_chars = set((ord("\n"), ord(" "), ord("\t"), ord("\r"), ord("/")))
 
     def _consume_whitespace(self) -> None:
         """
         Consume whitespace and comments.
         """
-        self._consume_regex(self._whitespace_regex)
+        first = self.input.slice(self.pos, self.pos + 1)
+        if first is None:
+            return
+        if ord(first) in self._first_whitespace_chars:
```

Comment:
Is the `ord` necessary? Does it make it significantly faster than directly checking character equality?

Reply:
I tried make filecheck with char in string, char in set, ord in set, and ord in set won.